common/source/docs/common-blheli32-passthru.rst

```diff
@@ -1,62 +1,158 @@
 
-In a similar fashion, normal output rotation direction can be reversed by setting :ref:`SERVO_BLH_RVMASK<SERVO_BLH_RVMASK>` without any changes needing to be made through ESC setup software (e.g. BLHeliSuite). This can also be used on ESCs with forward and reversed active operation, ie reversible ESCs, to set the "forward" direction's rotation.
+  .. image:: ../../../images/blheli-reversible-dshot.png
+    :target: ../_images/blheli-reversible-dshot.png
+    :width: 450px
 
 .. note:: Currently, ArduPilot only supports the use of reversible ESCs for Plane and Rover, not Copter.
 
-BLHeli32 ESC Telemetry
-----------------------
+.. _blheli32-esc-telemetry:
+
+ESC Telemetry
+-------------
+
+If the ESC has this capability, it allows monitoring and logging of performance data that previously required additional sensors (like power modules and RPM sensors). The detailed data provided by every ESC allows real-time decisions and individual ESC or motor performance tuning and failure analysis. Note that a given ESC may or may not have a specific sensor's data transmitted via telemetry. It is common for 4 in 1 escs to provide voltage and current sensors but not transmit the data via telemetry, but rather by direct connection to the autopilot. Check the ESC data sheet and connection information for details.
+ 
+.. note:: ArduPilot does not currently support the polling of the ESCs for telemetry data via throttle idle messages over the signal line in non DShot protocols.
+
+Connecting the ESCs Telemetry wire
+----------------------------------
+
+.. image:: ../../../images/dshot-pixhawk.jpg
+    :target: ../_images/dshot-pixhawk.jpg
+    :width: 600px
+
+Connect all ESC telemetry wires to a single serial port's RX pin on the autopilot (above diagram uses Serial5 as an example).  A pin or wire for ESC telemetry is pre-soldered on most BLHeli32 ESCs. If the wire isn't pre-soldered you will need to solder it yourself. CubePilot serial port pinsouts can be found :ref:`here <common-thecube-overview>`.
+
+Set the following parameters to enable BLHeli32 telemetry feedback to the autopilot's serial port:
+
+- :ref:`SERIALx_PROTOCOL <SERIAL5_PROTOCOL>` 16 (= ESC telemetry) where "x" is the autopilot serial port number connected to the ESCs telemetry wire.  The mapping between serial port numbering and UART physical ports for you autopilot should be documented in its description page linked :ref:`here <common-autopilots>`.
+
+- Set :ref:`SERVO_BLH_AUTO <SERVO_BLH_AUTO>` to 1 to automatically enable pass-through on all outputs configured as motors (e.g. :ref:`SERVOx_FUNCTION <SERVO9_FUNCTION>` = "Motor1", "Motor2", etc) for multicopters and quadplanes or throttle (e.g. those with :ref:`SERVOx_FUNCTION <SERVO9_FUNCTION>` set to 70 ("throttle"), 73 ("throttle left") or 74 ("throttle right")) on rovers.  For most multicopters, quadplanes and rovers this will do the right thing but for planes, you will need to further specify the used outputs as follows:
```

Comment:
auto is not necessary for telemetry to work

Reply:
OK.  I guess if SERVO_BLH_AUTO is not required then SERVO_BLH_MASK, SERVO_BLH_OTYPE are also not required.

Users should have already followed instructions for Dshot so MOT_PWM_TYPE should already be set.